app/src/main/java/hr/foi/air/servicesync/backend/FirestoreService.kt

```diff
@@ -0,0 +1,60 @@
+package hr.foi.air.servicesync.backend
+
+import com.google.firebase.firestore.FirebaseFirestore
+
+class FirestoreService {
+    private val db = FirebaseFirestore.getInstance()
+
+    fun saveReservation(
+        companyId: String,
+        serviceName: String,
+        reservationDate: Long,
+        userId: String,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
+    ) {
+        val reservation = hashMapOf(
+            "companyId" to companyId,
+            "serviceName" to serviceName,
+            "reservationDate" to reservationDate,
+            "userId" to userId
+        )
+        val documentId = "$companyId-$userId-$reservationDate"
+
+        db.collection("reservations")
+            .document(documentId).set(reservation)
+            .addOnSuccessListener { onSuccess() }
+            .addOnFailureListener { onFailure(it) }
+    }
+
+    fun getAvailableTimeSlots(
+        companyId: String,
+        date: Long,
+        onSuccess: (List<Long>) -> Unit,
+        onFailure: (Exception) -> Unit
+    ) {
+        val startOfDay = date - (date % (24 * 60 * 60 * 1000)) // Početak dana
+        val endOfDay = startOfDay + (24 * 60 * 60 * 1000) - 1 // Kraj dana
```

Comment:
Remove comment, variable names explain what they store

Reply:
I agree with the colleague's comment, though it is not a pressing issue

Reply:
Noted for the next sprint, as we are in a hurry with this pull.